docs/2-architecture/readme.md

```diff
@@ -55,3 +55,100 @@ The main entry point of the protocol is `AragonCourt`, this component inherits f
+
+Notes: 
+- If there are no ongoing disputes any of the three dependencies can be swapped 
+- If there are ongoing disputes none of the three dependencies can be migrated 
+
+
+### 2.5.2. Voting
+
+| Relies on module |           On           |     Expected behavior     |
+|------------------|------------------------|---------------------------|
+| DisputeManager   | `create`               | Only the latest Dispute Manager should be able to create votes                              |
+| DisputeManager   | `commit`               | The Voting module needs to access the same Dispute Manager instance during a vote lifecycle |
+| DisputeManager   | `reveal`               | " |
+| DisputeManager   | `leak`                 | " |
+
+Notes: 
+- If there are ongoing disputes the Voting dependency cannot be changed
+- If a new Dispute Manager is deployed it can be pointed to the same Voting module 
+
+
+### 2.5.3. Treasury
+
+| Relies on module |           On           |     Expected behavior     |
+|------------------|------------------------|---------------------------|
+| DisputesManager  | `assign`               | Any active Dispute Manager should be able to assign token balances |
+
+Notes:
+- The Treasury module shouldn't reference any particular Dispute Manager instance, deploying a new Dispute Manager is not a problem
+
+
+### 2.5.4. Jurors Registry
+
+| Relies on module |           On           |    Expected behavior     |
+|------------------|------------------------|--------------------------|
+| DisputesManager  | `assignTokens`         | Any active Dispute Manager should be able to manage juror tokens |
+| DisputesManager  | `burnTokens`           | Any active Dispute Manager should be able to manage juror tokens |
+| DisputesManager  | `draft`                | Any active Dispute Manager should be able to manage juror tokens |
+| DisputesManager  | `clashOrUnlock`        | Any active Dispute Manager should be able to manage juror tokens |
+| DisputesManager  | `collectTokens`        | Any active Dispute Manager should be able to manage juror tokens |
+| DisputesManager  | `lockWithdrawals`      | Any active Dispute Manager should be able to manage juror tokens |
+
+Notes:
+- The Jurors Registry shouldn't reference any particular Dispute Manager instance, deploying a new Dispute Manager is not a problem
+
+
+### 2.5.5. Subscriptions
+
+| Relies on module |           On           |    Expected behavior     |
+|------------------|------------------------|--------------------------|
+| JurorsRegistry   | `getJuror`             | Each period should work always with the same Jurors Registry instance |
+| JurorsRegistry   | `getJurorShare`        | " |
+| JurorsRegistry   | `claimFees`            | " |
+| JurorsRegistry   | `ensurePeriodBalance`  | " |
+
+Notes:
+- The Subscriptions module could checkpoint the Jurors Registry instance per period
+- Otherwise, it can trust that the Jurors Registry always reflects the same status among all its versions
```

Comment:
Ah, we can leave this as an issue, but it'd also be worth also adding the new events / external interfaces to the docs!

Reply:
Not sure what do you mean here

Reply:
Let's discuss in a separate issue! https://github.com/aragon/protocol/issues/22